Lib/ntpath.py

```diff
@@ -475,19 +470,17 @@ def expandvars(path):
                 pathlen = len(path)
                 try:
                     index = path.index(percent)
+                    var = path[:index]
+                    if environ is None:
+                        value = os.fsencode(os.environ[os.fsdecode(var)])
+                    else:
+                        value = environ[var]
+                    res += value
                 except ValueError:
                     res += percent + path
                     index = pathlen - 1
-                else:
-                    var = path[:index]
-                    try:
-                        if environ is None:
-                            value = os.fsencode(os.environ[os.fsdecode(var)])
-                        else:
-                            value = environ[var]
-                    except KeyError:
-                        value = percent + var + percent
-                    res += value
+                except KeyError:
+                    res += percent + var + percent
```

Comment:
This change reduces readability and safety of the code a little. Currently, the `try` block is very small, so it is clear to the reader which part of the code might raise `ValueError`. With this change, there's a much larger block of code inside the `try` statement, so it's much harder for the reader to understand which statement exactly is the one that might raise `ValueError`. That also introduces a risk of `ValueError` being unintentionally suppressed if it was unexpectedly raised by some other statement in the block, such as `os.fsdecode()` or `os.fsencode()`.

How much does this specific change speed things up?

Reply:
Nothing measureable.